dptb/tests/test_trainer_hardening.py

```
@@ -0,0 +1,386 @@
+    # ---- fresh start ----
+    ds = make_ds()
+    torch.manual_seed(0)
+    model = build_model(None, {k: (dict(v) if isinstance(v, dict) else v) for k, v in mo.items()}, dict(common))
+    ds.E3statistics(model=model)                       # data-based head calibration (gains stay 1.0)
+    trainer = Trainer(train_options=dict(train_opts), common_options=dict(common), model=model, train_datasets=ds)
+    assert trainer.ema is not None and len(trainer.optimizer.param_groups) > 1
+    for _ in range(2):
+        for batch in trainer.train_loader:
+            trainer.iteration(batch)
+            break
+    ckpt_dir = tmp_path / "ckpt"; ckpt_dir.mkdir()
```

Ruff (0.15.21)

[error] Multiple statements on one line (semicolon)

(E702)